pandajedi/jedibrokerage/AtlasBrokerUtils.py

```diff
@@ -642,6 +643,54 @@ def getOkNgArchList(task_spec):
     return (None, None)
 
 
+# get to-running rate of sites from various resources
+CACHE_SiteToRunRateStats = {}
+def getSiteToRunRateStats(tbIF, vo, time_window=21600, cutoff=300):
+    # initialize
+    ret_val = False
+    ret_map = {}
+    # timestamps
+    current_time = datetime.datetime.utcnow()
+    starttime_max = current_time - datetime.timedelta(seconds=cutoff)
+    starttime_min = current_time - datetime.timedelta(seconds=time_window)
+    # rounded with 10 minutes
+    starttime_max_rounded = starttime_max.replace(minute=starttime_max.minute//10*10, second=0, microsecond=0)
+    starttime_min_rounded = starttime_min.replace(minute=starttime_min.minute//10*10, second=0, microsecond=0)
+    real_interval_hours = (starttime_max_rounded - starttime_min_rounded).total_seconds()/3600
+    # cache key
+    cache_key = (starttime_min_rounded, starttime_max_rounded)
+    # condition of query
+    if cache_key in CACHE_SiteToRunRateStats \
+        and current_time >= CACHE_SiteToRunRateStats[cache_key]['exp']:
+        # query from local cache
+        ret_val = True
+        ret_map = CACHE_SiteToRunRateStats[cache_key]['data']
+    else:
+        if False:
+        # query from DB cache
+        # cached_data = tbIF.getCachedData(key='SiteToRunRateStats')
```

Comment:
If there exists a table in PanDA DB acting like harvester cache_table where arbitrary data can be shared across all nodes, this method can be implemented.